Syntax error in generated MSW when API returns polymorphic type

<!-- Thanks for reporting an issue! Please fill out the blanks below. -->

## What are the steps to reproduce this issue?

1. Use the following OpenAPI specification as input
```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "API",
    "version": "V1"
  },
  "paths": {
    "/flags": {
      "get": {
        "summary": "Get all flags.",
        "description": "Get all flags.",
        "operationId": "getFlags",
        "responses": {
          "200": {
            "description": "A list of flags with details including key, value and data type",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "oneOf": [
                      {
                        "$ref": "#/components/schemas/BooleanFlag"
                      },
                      {
                        "$ref": "#/components/schemas/JsonFlag"
                      },
                      {
                        "$ref": "#/components/schemas/NullFlag"
                      },
                      {
                        "$ref": "#/components/schemas/NumberFlag"
                      },
                      {
                        "$ref": "#/components/schemas/StringFlag"
                      }
                    ]
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "BooleanFlag": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "description": "Represent a feature flag with a boolean value.",
        "allOf": [
          {
            "$ref": "#/components/schemas/FlagDetails"
          },
          {
            "type": "object",
            "properties": {
              "value": {
                "type": "boolean"
              }
            }
          }
        ]
      },
      "FlagDetails": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "properties": {
          "value": {
            "type": "object"
          },
          "key": {
            "type": "string"
          },
          "type": {
            "type": "string",
            "enum": [
              "BOOLEAN",
              "STRING",
              "NUMBER",
              "JSON",
              "NULL"
            ]
          }
        },
        "discriminator": {
          "propertyName": "type"
        }
      },
      "JsonFlag": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "description": "Represent a feature flag with a raw JSON value.",
        "allOf": [
          {
            "$ref": "#/components/schemas/FlagDetails"
          },
          {
            "type": "object",
            "properties": {
              "value": {
                "type": "string"
              }
            }
          }
        ]
      },
      "NullFlag": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "description": "Represent a feature flag with no value.",
        "allOf": [
          {
            "$ref": "#/components/schemas/FlagDetails"
          }
        ]
      },
      "NumberFlag": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "description": "Represent a feature flag with numeric value.",
        "allOf": [
          {
            "$ref": "#/components/schemas/FlagDetails"
          },
          {
            "type": "object",
            "properties": {
              "value": {
                "type": "number"
              }
            }
          }
        ]
      },
      "StringFlag": {
        "required": [
          "key",
          "type",
          "value"
        ],
        "type": "object",
        "description": "Represent a feature flag with string value.",
        "allOf": [
          {
            "$ref": "#/components/schemas/FlagDetails"
          },
          {
            "type": "object",
            "properties": {
              "value": {
                "type": "string"
              }
            }
          }
        ]
      }
    }
  }
}
```
2. Configure Orval to generate mocks (`output.mock: true`)
3. Run Orval

## What happens?

As of version 6.9.X, the mock generator function is missing curly braces around the object declaration, making the generated statement invalid
```typescript
export const getGetFlagsMock = () => (Array.from({ length: faker.datatype.number({ min: 1, max: 10 }) }, (_, i) => i + 1).map(() => (faker.helpers.arrayElement([value: {}, key: faker.random.word(), type: faker.helpers.arrayElement(['BOOLEAN','STRING','NUMBER','JSON','NULL']),,value: {}, key: faker.random.word(), type: faker.helpers.arrayElement(['BOOLEAN','STRING','NUMBER','JSON','NULL']),,value: {}, key: faker.random.word(), type: faker.helpers.arrayElement(['BOOLEAN','STRING','NUMBER','JSON','NULL']),value: {}, key: faker.random.word(), type: faker.helpers.arrayElement(['BOOLEAN','STRING','NUMBER','JSON','NULL']),,value: {}, key: faker.random.word(), type: faker.helpers.arrayElement(['BOOLEAN','STRING','NUMBER','JSON','NULL']),]))))
```

## What were you expecting to happen?

In versions before 6.9.X, the generator function was correct
```typescript
export const getGetFlagsMock = () =>
  [...Array(faker.datatype.number({ min: 1, max: 10 }))].map(() =>
    faker.random.arrayElement([
      {
        value: {},
        key: faker.random.word(),
        type: faker.random.arrayElement([
          'BOOLEAN',
          'STRING',
          'NUMBER',
          'JSON',
          'NULL',
        ]),
      },
      {
        value: {},
        key: faker.random.word(),
        type: faker.random.arrayElement([
          'BOOLEAN',
          'STRING',
          'NUMBER',
          'JSON',
          'NULL',
        ]),
      },
      {
        value: {},
        key: faker.random.word(),
        type: faker.random.arrayElement([
          'BOOLEAN',
          'STRING',
          'NUMBER',
          'JSON',
          'NULL',
        ]),
      },
      {
        value: {},
        key: faker.random.word(),
        type: faker.random.arrayElement([
          'BOOLEAN',
          'STRING',
          'NUMBER',
          'JSON',
          'NULL',
        ]),
      },
      {
        value: {},
        key: faker.random.word(),
        type: faker.random.arrayElement([
          'BOOLEAN',
          'STRING',
          'NUMBER',
          'JSON',
          'NULL',
        ]),
      },
    ]),);
```

## Any logs, error output, etc?

```sh
> orval --clean && echo 'Running ESLint on generated files, this may take a moment...' && eslint --config .eslintrc.js src/generated/ --fix

🍻 Start orval v6.9.6 - A swagger client generator for typescript
api: Cleaning output folder
🎉 api - Your OpenAPI spec has been converted into ready to use orval!
Running ESLint on generated files, this may take a moment...

/.../src/generated/api/default/default.msw.ts
  14:166  error  Parsing error: ',' expected

✖ 1 problem (1 error, 0 warnings)

```
```txt
/.../src/generated/api/default/default.msw.ts
Error:(14, 162) TS2304: Cannot find name 'value'.
Error:(14, 173) TS2304: Cannot find name 'key'.
Error:(14, 199) TS2304: Cannot find name 'type'.
Error:(14, 278) TS2304: Cannot find name 'value'.
Error:(14, 289) TS2304: Cannot find name 'key'.
Error:(14, 315) TS2304: Cannot find name 'type'.
Error:(14, 394) TS2304: Cannot find name 'value'.
Error:(14, 405) TS2304: Cannot find name 'key'.
Error:(14, 431) TS2304: Cannot find name 'type'.
Error:(14, 509) TS2304: Cannot find name 'value'.
Error:(14, 520) TS2304: Cannot find name 'key'.
Error:(14, 546) TS2304: Cannot find name 'type'.
Error:(14, 625) TS2304: Cannot find name 'value'.
Error:(14, 636) TS2304: Cannot find name 'key'.
Error:(14, 662) TS2304: Cannot find name 'type'.
Error:(14, 167) TS1005: ',' expected.
Error:(14, 176) TS1005: ',' expected.
Error:(14, 203) TS1005: ',' expected.
Error:(14, 283) TS1005: ',' expected.
Error:(14, 292) TS1005: ',' expected.
Error:(14, 319) TS1005: ',' expected.
Error:(14, 399) TS1005: ',' expected.
Error:(14, 408) TS1005: ',' expected.
Error:(14, 435) TS1005: ',' expected.
Error:(14, 514) TS1005: ',' expected.
Error:(14, 523) TS1005: ',' expected.
Error:(14, 550) TS1005: ',' expected.
Error:(14, 630) TS1005: ',' expected.
Error:(14, 639) TS1005: ',' expected.
Error:(14, 666) TS1005: ',' expected.
```


## Any other comments?
N/A

## What versions are you using?

**Operating System:** Arch Linux 5.19.3-arch1-1
**Package Version:** Orval: 6.9.6, msw: 0.45.0
**Browser Version:** Firefox 103.0.2 (64-bit)

--- a/src/core/getters/combine.mock.ts
+++ b/src/core/getters/combine.mock.ts
@@ -78,7 +78,10 @@ export const combineSchemasMock = ({
         ? `${resolvedValue.value.slice(0, -1)},${itemResolvedValue.value}}`
         : resolvedValue.value;
 
-    if (!index && !combine) {
+    const isObjectBounds =
+      !combine || (combine.separator === 'oneOf' && separator === 'allOf');
+
+    if (!index && isObjectBounds) {
       if (resolvedValue.enums || separator === 'oneOf') {
         if (arr.length === 1) {
           return `faker.helpers.arrayElement([${value}])`;
@@ -99,7 +102,7 @@ export const combineSchemasMock = ({
       }
       return `${acc}${value}${
         itemResolvedValue?.value ? `,${itemResolvedValue.value}` : ''
-      }${!combine ? '}' : ''}`;
+      }${isObjectBounds ? '}' : ''}`;
     }
     if (!value) {
       return acc;

--- a/tests/configs/react-query.config.ts
+++ b/tests/configs/react-query.config.ts
@@ -268,4 +268,16 @@ export default defineConfig({
       target: '../specifications/petstore.yaml',
     },
   },
+  polymorphic: {
+    output: {
+      target: '../generated/react-query/polymorphic/endpoints.ts',
+      schemas: '../generated/react-query/polymorphic/model',
+      client: 'react-query',
+      mock: true,
+      headers: true,
+    },
+    input: {
+      target: '../specifications/polymorphic.yaml',
+    },
+  },
 });

--- a/tests/specifications/polymorphic.yaml
+++ b/tests/specifications/polymorphic.yaml
@@ -1,0 +1,50 @@
+openapi: 3.0.1
+info:
+  title: API
+  version: V1
+paths:
+  /demo:
+    get:
+      operationId: getPolymorphicResponse
+      responses:
+        '200':
+          description: ''
+          content:
+            application/json:
+              schema:
+                type: object
+                oneOf:
+                  - $ref: '#/components/schemas/DescendantOne'
+                  - $ref: '#/components/schemas/DescendantTwo'
+components:
+  schemas:
+    DescendantOne:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/ParentType'
+        - type: object
+          properties:
+            value:
+              type: boolean
+    DescendantTwo:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/ParentType'
+        - type: object
+          properties:
+            value:
+              type: string
+    ParentType:
+      type: object
+      properties:
+        value:
+          type: object
+        key:
+          type: string
+        type:
+          type: string
+          enum:
+            - BOOLEAN
+            - STRING
+      discriminator:
+        propertyName: type

